frontend/src/test/java/com/google/sps/InMemoryTripModelStorageTest.java

```diff
@@ -0,0 +1,266 @@
+    storage.addTrip(LONDON_TRIP1);
+    Assert.assertThrows(TripAlreadyExistsException.class, () -> 
+        storage.addTrip(LONDON_TRIP_DUPLICATE_ID));
+    
+  }
+
+  /**
+   * test that adding a different TripModel object does not cause an exception
+   * to be thrown
+   */
+  @Test
+  public void addTrip_tripNotInStorage_exceptionNotThrown() 
+      throws TripAlreadyExistsException {
+    InMemoryTripModelStorage storage = new InMemoryTripModelStorage();
+    storage.addTrip(LONDON_TRIP1);
+    storage.addTrip(LONDON_TRIP2);
+  }
+
+  /**
+   * test that removing a trip from empty storage causes an exception to be 
+   * thrown
+   */
+  @Test
+  public void removeTrip_emptyStorage_thowsException() 
+      throws TripNotFoundException {
+    InMemoryTripModelStorage storage = new InMemoryTripModelStorage();
+    Assert.assertThrows(TripNotFoundException.class, () -> 
+        storage.removeTrip(LONDON_TRIP1.id()));
+  }
+
+  /**
+   * test that removing a trip from storage when there is not a TripModel object 
+   * with the given tripId in storage causes an exception to be thrown
+   */
+  @Test
+  public void removeTrip_tripNotInStorage_thowsException() 
+      throws TripAlreadyExistsException, TripNotFoundException {
+    InMemoryTripModelStorage storage = new InMemoryTripModelStorage();
+    storage.addTrip(LONDON_TRIP1);
+    Assert.assertThrows(TripNotFoundException.class, () -> 
+        storage.removeTrip(LONDON_TRIP1_2022.id()));
+  }
+
+  /**
+   * test that getting a trip from empty storage causes an exception to be 
+   * thrown
+   */
+  @Test
+  public void getTrip_emptyStorage_thowsException() throws TripNotFoundException {
+    InMemoryTripModelStorage storage = new InMemoryTripModelStorage();
+    Assert.assertThrows(TripNotFoundException.class, () -> 
+        storage.getTrip(LONDON_TRIP1.id()));
+  }
+
+  /**
+   * test that getting a trip from storage when there is not a TripModel object 
+   * with the given tripId in storage causes an exception to be thrown
+   */
+  @Test
+  public void getTrip_tripNotInStorage_thowsException() 
+      throws TripAlreadyExistsException, TripNotFoundException {
+    InMemoryTripModelStorage storage = new InMemoryTripModelStorage();
+    storage.addTrip(LONDON_TRIP1);
+    Assert.assertThrows(TripNotFoundException.class, () -> 
+        storage.getTrip(LONDON_TRIP1_2022.id()));
+  }
+
+  /**
+   * test that after adding a TripModel object to storage, getting the TripModel 
+   * object with the same tripId returns the same TripModel object as the added
+   * object
+   */
+  @Test
+  public void addTrip_getTrip_addedTripEqualsGottenTrip() 
+      throws TripAlreadyExistsException, TripNotFoundException {
+    InMemoryTripModelStorage storage = new InMemoryTripModelStorage();
+    storage.addTrip(LONDON_TRIP1);
+    storage.addTrip(LONDON_TRIP2);
+    TripModel gottenTrip = storage.getTrip(LONDON_TRIP2.id());
+    Assert.assertEquals(LONDON_TRIP2, gottenTrip);
```

Comment:
I strongly encourage you to switch to [Truth](https://truth.dev/) for these assertions. This results in more readable test code and error messages. The fluent style makes it clear which of the inputs you're testing and what the expected value is. For example, this might be translated as such:

```java
// Optionally, at top of file. Otherwise, you'd invoke it below as Truth.assertThat().
import static com.google.common.truth.Truth.assertThat;

// Within test case
assertThat(gottenTrip).isEqualTo(LONDON_TRIP2)
```